api/router/devices.go

```diff
@@ -0,0 +1,75 @@
+package router
+
+import (
+	"fmt"
+	"errors"
+	"github.com/TheThingsNetwork/ttn/api/models"
+	"github.com/TheThingsNetwork/ttn/api/auth"
+	"github.com/gin-gonic/gin"
+)
+
+func getDevice(c *gin.Context) {
+
+	token,  _ := c.Get("token")
+	appEUI, _ := c.Get("appEUI")
+	devEUI, _ := c.Get("devEUI")
+
+	dev, err := models.DeviceInfo(token.(auth.Token), appEUI.(models.EUI), devEUI.(models.EUI))
+	if err != nil {
+		c.AbortWithError(404, err)
+		return
+	}
+
+	c.JSON(200, dev)
+}
+
+type registerDeviceFields struct {
+	AppEUI  models.EUI  `json:"application"`
+	DevEUI  models.EUI  `json:"device"`
+	AppKey  models.SKey `json:"app_secret" binding:"required"`
+	NwkKey  models.SKey `json:"nwk_secret"`
+	Type    string      `json:"type" binding:"required"`
+	Address []byte      `json:"address"`
+}
+
+func registerDevice(c *gin.Context) {
```

Comment:
Use different endpoints for OTAA and personalized, also with different JSON contracts. Fields are very different